std/conv.d

```diff
@@ -3992,6 +3992,8 @@ if (is(UT == Unqual!UT))
 //emplace helper functions
 private void emplaceInitializer(T)(ref T chunk) @trusted pure nothrow
 {
+    static if (is(T == class) && isAbstractClass!T)
+        static assert(0, T.stringof ~ " is abstract and it can't be emplaced");
```

Comment:
`emplaceInitializer()` is a private helper function, so it's not a good place to add the check which affects emplace/emplaceRef behavior. I think it's better to move this compile time check to `emplaceRef`.

And, you can rewrite the check to a single `static assert`:
```d
static assert(!is(T == class) || !isAbstractClass!T,
    T.stringof ~ " is abstract and it can't be emplaced");
```